### PR TITLE
Add recipes for CSS properties

### PR DESCRIPTION
**Pull request by wbamberg**
This PR adds recipes for CSS properties, as discussed in https://github.com/mdn/stumptown-content/issues/396.

Apart from the substantive additions there's some rework, which I put into separate commits to (I hope) make it clearer.

Currently we have a spec called "javascript-linter-spec.md", but I wanted to have just one spec across all domains, since they share so many ingredients. But since we probably link to "javascript-linter-spec.md" from various spots, I wanted to keep that document around. So:

* https://github.com/mdn/stumptown-content/commit/2421ad415d192fe896d1be3158b7546dd633276d creates a new doc "linter-spec.md", starting out with the same content as "javascript-linter-spec.md"
* https://github.com/mdn/stumptown-content/commit/1c0553d933d638f4845d9f56ffdb4b4f3791830d updates "javascript-linter-spec.md" to be just a pointer to the new document
* https://github.com/mdn/stumptown-content/commit/16cecd8878d5c16f4fd8ba2aecc707635a50d499 makes "linter-spec.md" not be JS-specific
*  https://github.com/mdn/stumptown-content/commit/7c291cef72114941f262d91252f0dba18fb6685e alphabetises the ingredient descriptions in "linter-spec.md"
* https://github.com/mdn/stumptown-content/commit/d35b2328f938e486938be65671d451d7b8d04ce3 and https://github.com/mdn/stumptown-content/commit/6adbdfe686db29bb8f8aad7e47e00dd14e630462 actually add the new recipes and descriptions

There are two recipes, one for normal CSS properties and one for shorthand properties, and three new ingredients:

* `data.constituent_properties`
* `data.formal_definition`
* `data.formal_syntax`

## Constituent properties

`data.constituent_properties` is only for shorthand properties, and is only allowed to list the properties the shorthand contains.

## Formal definition and syntax

Normal properties get two extra ingredients:
* `data.formal_definition`: this contains the `{{CSSInfo}}` call
* `data.formal_syntax`: this contains the `{{CSSSyntax}}` call

After some discussion in https://github.com/mdn/stumptown-content/issues/396 and also in https://github.com/mdn/stumptown-content/issues/440#issuecomment-628365377, I'm proposing to make these things be separate H2 headings, rather than have one ingredient "Formal definition and syntax". This is because it seems plausible to expect that other CSS things might want formal syntax but not formal definition.

We could break the linkage between H2 headings and ingredients, and allow ingredients to live anywhere in the outline. But having thought a bit about this I think it's more trouble than it's worth. Suppose CSS properties wanted a structure like:

```
## Formal definition and syntax
    ### Formal definition
    ### Formal syntax
```

Then suppose CSS selectors wanted to keep formal syntax, but not formal definition. Then they might want it to live in a top-level H2:

```
## Formal syntax
```

Then the linter would need to apply different rules for the same ingredient, depending on the recipe it's living in. This breaks the rule that the definition of an ingredient is independent of the recipes that uses it, and this rule seems important for keeping things simple. It means that once you understand the definition, you can use it anywhere.

### CODE REVIEW

File: project-docs/linter-spec.md [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+Each example may be a simple example or a live sample. If the example contains a call to the `{{EmbedLiveSample}}` macro, it is a live sample. Otherwise it is a simple example.
+
+Live samples must satisfy certain additional constraints:
+
+- The call to `{{EmbedLiveSample}}` must be the last element in the example (the only exception is that text nodes consisting only of newlines may follow the `{{EmbedLiveSample}}` call). This implies that there may only be one live sample in a given H3 section, and that the iframe showing the output must appear last in the rendered page, after any code or explanatory prose.
+- The ID parameter passed to `{{EmbedLiveSample}}` must match the ID of the example's H3 heading. This implies that the example is not allowed to contain any code blocks that are not included in the live sample.
+
+#### data.formal_definition
+
+To satisfy this ingredient a page must have a section demarcated by `H2#Formal_definition`.It must contain only a call to the `{{CSSInfo}}` macro.
```
Comment (ddbeck) [inline project-docs/linter-spec.md:155]:
```suggestion
To satisfy this ingredient a page must have a section demarcated by `H2#Formal_definition`. It must contain only a call to the `{{CSSInfo}}` macro.
```

File: project-docs/linter-spec.md [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+#### data.formal_syntax
+
+To satisfy this ingredient a page must have a section demarcated by `H2#Formal_syntax`.It must contain only a call to the `{{CSSSyntax}}` macro.
```
Comment (ddbeck) [inline project-docs/linter-spec.md:159]:
```suggestion
To satisfy this ingredient a page must have a section demarcated by `H2#Formal_syntax`. It must contain only a call to the `{{CSSSyntax}}` macro.
```

**Review by ddbeck — APPROVED** [review]
I noticed the two tiniest of formatting problems, but other than that, this looks really good. Thanks for putting in a lot of good work on this, Will.

(This was suppose to be "Changes requested" but I goofed. Please consider this approved pending the suggestions below.)

**Review by ddbeck — APPROVED** [review]

**Comment by ddbeck** [comment]
I wasn't sure if @Elchi3 also wanted a look at this first, but this is good to merge, if not.

File: project-docs/linter-spec.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+- [javascript-namespace](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-namespace.yaml)
+- [javascript-constructor](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-constructor.yaml)
+- [javascript-method](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-method.yaml)
+- [javascript-property](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-property.yaml)
+- [javascript-error](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-error.yaml)
+- [javascript-language-feature](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-language-feature.yaml)
+
+## Recipe ingredients
+
+In total these recipes use the following ingredients:
```
Comment (Elchi3) [inline project-docs/linter-spec.md:47]: Above you only mention JS recipes but then here you list all ingredients for all recipes. 
Should we also list CSS recipes?

**Review by Elchi3 — COMMENTED** [review]
Looks good to me, thanks Will! 
Just one comment that isn't blocking but that caught me while reading.

File: project-docs/linter-spec.md (hunk 3 above)
Comment (wbamberg) [reply project-docs/linter-spec.md:52]: Oops, that should be fixed now.

**Review by Elchi3 — APPROVED** [review]

**Pull request merged**